docs/user-guide/docker-compose.md

```diff
@@ -0,0 +1,380 @@
+
+```bash
+# restart the container to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the same diagnostics endpoint as the previous step
+curl -I localhost:8080/ambassador/v0/diag/
+
+# the response code should be 404
+HTTP/1.1 404 Not Found
+date: Fri, 17 Aug 2018 21:18:25 GMT
+server: envoy
+content-length: 0
+```
+
+Feel free to re-enable the diagnostics endpoint.
+
+## 3. Add a route mapping
+
+Now that we have demonstrated that we can modify the configuration let's add a mapping to route to `http://demo.getambassador.io/qotm/` service.
+
+Create a new file `config/mapping-qotm.yaml` with these contents:
+
+```yaml
+---
+apiVersion: ambassador/v0
+kind:  Mapping
+name:  qotm_mapping
+prefix: /qotm/
+rewrite: /qotm/
+service: demo.getambassador.io
+```
+
+Once again, restart ambassador and test the new mapping:
+
+```bash
+# restart the container to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service
+curl localhost:8080/qotm/quote
+
+# the response body should be a json object with a quote
+{
+  "hostname": "qotm-3716059461-47tnl",
+  "ok": true,
+  "quote": "The light at the end of the tunnel is interdependent on the relatedness of motivation, subcultures, and management.",
+  "time": "2018-08-17T21:29:24.690950",
+  "version": "1.3"
+}
+```
+
+## 3. Add a route mapping to an internal service
+
+While routing to an external service is useful, more often than not our Docker Compose environment will contain a number of services that need routing.
+
+### Add the qotm service to the docker-compose.yaml file
+
+Edit the `docker-compose.yaml` file and add a new `qotm` service. It should now look like this:
+
+```yaml
+version: '3.5'
+
+services:
+  ambassador:
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
+    volumes:
+    # mount a volume where we can inject configuration files
+    - ./config:/ambassador/ambassador-config
+  qotm:
+    image: datawire/qotm:1.3
+    ports:
+    - 5000
+```
+
+### Update the mapping-qotm.yaml file to route to our internal qotm service
+
+Edit the `config/mapping-qotm.yaml` file and modify the `service` and `rewrite` field. It should now look like this:
+
+```yaml
+---
+apiVersion: ambassador/v0
+kind:  Mapping
+name:  qotm_mapping
+prefix: /qotm/
+# remove the qotm prefix when talking to the qotm service
+rewrite: /
+# change the `service` parameter to the name of our service with the port
+service: qotm:5000
+```
+
+### Restart Ambassador and test
+
+Re-run the same test as in the previous section to ensure the route works as before. This time we will need to bring up the new service first.
+
+```bash
+# start all new containers (eg. qotm)
+docker-compose up -d
+
+# restart the container to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service
+curl localhost:8080/qotm/quote
```

Comment:
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.